base/utils/R/ud_convert.R

```diff
@@ -3,15 +3,21 @@
 ##' Unit conversion to replace the now-unmaintained `udunits2::ud.convert`
 ##' @author Chris Black
 ##' 
-##' @param x numeric vector
-##' @param u1 string parseable as the units in which `x` is provided
+##' @param x vector of class "numeric" or "difftime"
+##' @param u1 string parseable as the units in which `x` is provided.  If `x` is
+##'   class "difftime", then `u1` is not actually used.  However, it still needs
+##'   to be supplied and needs to be convertible to `u2` for consistency.
 ##' @param u2 string parseable as the units to convert to
 ##' 
 ##' @return numeric vector with values converted to units in `u2`
 ##' @export
 ud_convert <- function(x, u1, u2) {
   stopifnot(units::ud_are_convertible(u1, u2))
-  x1 <- units::set_units(x, value = u1, mode = "standard")
+  if(inherits(x, "difftime")) {
+    x1 <- units::as_units(x)
```

Comment:
I'm nervous about ignoring `u1` here -- I don't know how often units(x) and u1 will mismatch in practice, but when they do it means at a minimum that the user's expectations are wrong and the earlier we alert them the better.

Maybe something like...?
```suggestion
    stopifnot(units::as_units(units(x)) == units::as_units(u1))
    x1 <- units::as_units(x)
```